src/Kubernetes.Controller/Converters/YarpParser.cs

```diff
@@ -208,6 +209,10 @@ private static YarpIngressOptions HandleAnnotations(YarpIngressContext context,
             // YamlDeserializer does not support IReadOnlyList<string> in RouteHeader for now, so we use RouteHeaderWapper to solve this problem.
             options.RouteHeaders = YamlDeserializer.Deserialize<List<RouteHeaderWapper>>(routeHeaders).Select(p => p.ToRouteHeader()).ToList();
         }
+        if (annotations.TryGetValue("yarp.ingress.kubernetes.io/route-order", out var routeOrder)&& int.TryParse(routeOrder,out var order ))
+        {
+            options.Order = order;
```

Comment:
```suggestion
        if (annotations.TryGetValue("yarp.ingress.kubernetes.io/route-order", out var routeOrder))
        {
            options.Order = int.Parse(routeOrder, CultureInfo.InvariantCulture);
```
Other annotations don't seem to do "try" parsing of provided values